bep032tools/generator/tests/test_BEP032Generator.py

```diff
@@ -184,7 +184,36 @@ def test_generate_example_structure(self):
     def doCleanups(self):
         initialize_test_directory(clean=True)
 
+class Test_FolderGeneration(unittest.TestCase):
+
+    def setUp(self):
+        self.test_dir = Path(initialize_test_directory(clean=True))
+        self.test_dir = self.test_dir / 'generateTest'
+        csv_filename = generate_simple_csv_file()
+        self.csv_file = csv_filename
+
+    def test_generate_folder(self):
+        """
+        When the folder did not exist, the generation did not work properly.
+        Now if the folder does not exist, the function creates the folder before the generation function is launched
+        """
+        test_generate = False
+        BEP032Data.generate_struct(self.csv_file, self.test_dir)
+        if os.path.isdir(self.test_dir):
+            test_generate = True
+        self.assertTrue(test_generate)
+
+    def test_no_duplicate_folder_generated(self):
+        """
+        Checks that there are no duplicates when generating folders.
+        If a duplicate exists and has the same name as the parent folder, the function returns an error
+        """
+        BEP032Data.generate_struct(self.csv_file, self.test_dir)
+        generation = False
+        if os.path.isdir(self.test_dir) and os.path.basename(self.test_dir) not in os.listdir(self.test_dir):
```

Comment:
The check for the generation of the root folder we already covered in the other test function, so we do not need to reimplement that here. Also you know exactly the filename you want to check for, so you don't need to iterate all existing folders.
```suggestion
    root_name = self.test_dir.name
        if  not (self.test_dir / root_name).exists()):
```